### PR TITLE
Enabling Translation

### PR DESCRIPTION
**Pull request by vjeux**
This is using @lipis content as base for this post

### CODE REVIEW

File: content/blog/enabling-translations/index.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+The only external library we're using is [i18next-browser-languagedetector](https://github.com/i18next/i18next-browser-languageDetector) which has a lot of great heuristics to figure out what language the user is currently using.
+
+Finally the last piece of the puzzle is to be able to change the language. We opted for a simple `<select>` element.
+
+![Language selector](language-selector.png)
+
+## Enter Crowdin
+
+At that point, if you wanted to contribute, you need to edit one of those JSON files on GitHub and send a pull request. While this process works, it was tedious to manage. We started adopting [Crowdin](https://crowdin.com) which dramatically improved the whole process.
```
Comment (dwelle) [inline content/blog/enabling-translations/index.md:72]:
```suggestion
Previously, if you wanted to contribute, you needed to edit one of the JSON files on GitHub and send a pull request. While this process worked, it was tedious to manage. Instead, we moved to [Crowdin](https://crowdin.com) which dramatically improved the whole process.
```

File: content/blog/enabling-translations/index.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+The initial setup was easy, you create a new project on Crowdin, upload your `en.json` file and connect a GitHub account. Then, any time a translation is added, a pull request is created/updated with the json files being modified.
```
Comment (lipis) [inline content/blog/enabling-translations/index.md:74]:
```suggestion
The initial setup was easy, you create a new project on Crowdin, upload your `en.json` file and connect a GitHub account. Then, any time a translation is added, a pull request is created/updated with the JSON files being modified.
```

File: content/blog/enabling-translations/index.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  { lng: "en", label: "English", data: require("./locales/en.json") },
+  { lng: "bg-BG", label: "Български", data: require("./locales/bg-BG.json") },
+  { lng: "de-DE", label: "Deutsch", data: require("./locales/de-DE.json") },
+  // ...
+];
+```
+
+The `t` function is pretty simple, where it splits the path by `.` and looks up both the current and fallback languages.
+
+```js
```
Comment (lipis) [inline content/blog/enabling-translations/index.md:50]:
```suggestion
```ts
```

**Comment by lipis** [comment]
Update the screenshot maybe for the completion file: https://crowdin.com/project/excalidraw

File: content/blog/enabling-translations/index.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
+---
+title: Enabling Translations
+date: 2020-04-15
```
Comment (lipis) [inline content/blog/enabling-translations/index.md:3]:
```suggestion
date: 2020-04-16
```

File: content/blog/enabling-translations/index.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+The only external library we're using is [i18next-browser-languagedetector](https://github.com/i18next/i18next-browser-languageDetector) which has a lot of great heuristics to figure out what language the user is currently using.
+
+Finally the last piece of the puzzle is to be able to change the language. We opted for a simple `<select>` element.
+
+![Language selector](language-selector.png)
+
+## Enter Crowdin
+
+Previously, if you wanted to contribute, you had to edit one of the JSON files on GitHub and send a pull request. While this process worked, managing this was tedious enough to make us look for alternatives. We chose [Crowdin](https://crowdin.com) which dramatically improved the whole process.
```
Comment (lipis) [inline content/blog/enabling-translations/index.md:72]:
```suggestion
Previously, if you wanted to contribute, you had to edit one of the JSON files on GitHub and send a pull request. While this process worked, managing this was tedious enough to make us look for alternatives. We chose [Crowdin](https://crowdin.com/project/excalidraw) which dramatically improved the whole process.
```

File: content/blog/enabling-translations/index.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
+---
+title: Enabling Translations
+date: 2020-04-16
+---
+
+From the early days people asked for [Excalidraw](https://excalidraw.com) to be translated to other languages. Translation infrastructure and community maintenance have historically been a pain to maintain. Thankfully, with projects like [Crowdin](https://crowdin.com) and [i18next-browser-languagedetector](https://github.com/i18next/i18next-browser-languageDetector), Excalidraw is now translated in 20 languages and the whole process has been very low maintenance.
```
Comment (lipis) [inline content/blog/enabling-translations/index.md:6]:
```suggestion
From the early days people asked for [Excalidraw](https://excalidraw.com) to be translated to other languages. Translation infrastructure and community maintenance have historically been a pain to maintain. Thankfully, with projects like [Crowdin](https://crowdin.com/project/excalidraw) and [i18next-browser-languagedetector](https://github.com/i18next/i18next-browser-languageDetector), Excalidraw is now translated in 20 languages and the whole process has been very low maintenance.
```

File: content/blog/enabling-translations/index.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    "paste": "Paste",
+    "selectAll": "Select all",
+    "copy": "Copy",
+    "copyAsPng": "Copy to clipboard as PNG",
+    // ...
+  }
+}
+```
+
+We started using a fully fledged internalization library but unfortunately it was a massive dependency and caused multiple React re-renders on startup and added one long round-trip. So instead [we rolled our own](https://github.com/excalidraw/excalidraw/blob/0ffbde77ac3c10b04b05328a793e753ed33ed736/src/i18n.ts) in less than 100 lines of code.
```
Comment (lipis) [inline content/blog/enabling-translations/index.md:35]:
```suggestion
We started using a fully fledged internalization library but unfortunately it was a massive dependency and caused multiple React re-renders on startup and added one long round-trip. So instead [we rolled our own](https://github.com/excalidraw/excalidraw/blob/master/src/i18n.ts) in less than 100 lines of code.
```

**Pull request merged**